File: src/main/java/toker/warbandscripts/panel/entity/NoticeBoard.java
package toker.warbandscripts.panel.entity;

import com.fasterxml.jackson.annotation.JsonView;
import toker.warbandscripts.panel.entity.pk.NoticeBoardPK;

import javax.persistence.*;
import java.util.Collection;
import java.util.Objects;

@Entity
@Table(name = "notice_board")
@IdClass(NoticeBoardPK.class)
public class NoticeBoard {

    private Integer index;
    private Server server;
    private String name;
    private Collection<NoticeBoardAccess> accesses;
    private Collection<NoticeBoardEntry> entries;

    @Id
    @Column(name = "index", nullable = false)
    public Integer getIndex() {
        return index;
    }

    public void setIndex(Integer id) {
        this.index = id;
    }

    @Id
    @ManyToOne
    @JoinColumn(name = "server_id", referencedColumnName = "id", nullable = false)
    public Server getServer() {
        return server;
    }

    public void setServer(Server server) {
        this.server = server;
    }

    @Column(name = "name", nullable = false, length = 64)
    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;
        NoticeBoard that = (NoticeBoard) o;
        return Objects.equals(index, that.index) &&
                Objects.equals(name, that.name);
    }

    @Override
    public int hashCode() {
        return Objects.hash(index, name);
    }

    @OneToMany(mappedBy = "board")
    @JsonView(View.Accesses.class)
    public Collection<NoticeBoardAccess> getAccesses() {
        return accesses;
    }

    public void setAccesses(Collection<NoticeBoardAccess> accesses) {
        this.accesses = accesses;
    }

    @OneToMany(mappedBy = "board")
    @JsonView(View.Entries.class)
    public Collection<NoticeBoardEntry> getEntries() {
        return entries;
    }

    public void setEntries(Collection<NoticeBoardEntry> entries) {
        this.entries = entries;
    }

    public static class View {
        public static class Accesses {}
        public static class Entries {}
        public static class None {}
    }
}

File: src/main/java/toker/panel/entity/NoticeBoard.kt
package toker.panel.entity

import com.fasterxml.jackson.annotation.JsonView
import toker.panel.entity.NoticeBoard.View.Accesses
import toker.panel.entity.pk.NoticeBoardPK
import java.util.*
import javax.persistence.*

@Entity
@Table(name = "notice_board")
@IdClass(NoticeBoardPK::class)
class NoticeBoard(
        @Id
        @Column(name = "index", nullable = false)
        var index: Int? = null,

        @Id
        @ManyToOne
        @JoinColumn(name = "server_id", referencedColumnName = "id", nullable = false)
        var server: Server? = null,

        @Column(name = "name", nullable = false, length = 64)
        var name: String? = null
) {
    @OneToMany(mappedBy = "board")
    @JsonView(Accesses::class)
    var accesses: MutableSet<NoticeBoardAccess>? = null

    @OneToMany(mappedBy = "board")
    @JsonView(View.Entries::class)
    var entries: MutableSet<NoticeBoardEntry>? = null

    interface View {
        interface Accesses
        interface Entries
        interface None
    }

    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (javaClass != other?.javaClass) return false
        other as NoticeBoard
        if (index != other.index) return false
        if (server != other.server) return false
        return true
    }

    override fun hashCode(): Int {
        var result = index ?: 0
        result = 31 * result + (server?.hashCode() ?: 0)
        return result
    }
}